Automattic-Tracks-iOS/LogFile.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+public class LogFile {
+    var uuid: String
+    var url: URL
+
+    public init(url: URL, uuid: String = UUID().uuidString) {
+        self.url = url
+        self.uuid = uuid
+    }
+
+    var exists: Bool {
+         return FileManager.default.fileExists(atPath: url.path)
```

Comment:
I suggest not creating a coupling with `FileManager` from `LogFile`. Especially since this is a computed property, it'll make it impossible/very hard to test anything that depends on a `LogFile` that doesn't exist, or to use a different FileManager for that matter.

If you want to keep the `exists` property, I'd convert it to function and take the `FileManager` as parameter. Alternatively, you can take a `doesFileExist` function as a parameter which will further remove its coupling.

---

My suggestion would be to keep this just as a `data` class and move `exists` to somewhere else.

In general, (excuse the pseudo-code):

```
bool checkSomething(something: Something)
```

instead of:

```
class Something {
    bool checkSomething()
}
```

will make it less likely to create couplings. Just to be clear, the goal is not to put everything outside of classes, but the first option will make the developer think more in terms of parameters instead of depending on internal dependencies. Once someone is comfortable with this line of thinking, it's easier to decide whether something is better off as a property/class-function or an external function.

The way I like to go about this is to _mostly_ separate my classes to 2 types. One type is just data classes that don't contain any logic and the other one is implementation oriented one. In the implementation oriented one, the more you work with parameters and interfaces, the easier time you'll have testing them. For data classes you won't even need to write any tests, so that's a bonus.

---

Finally, the reason that I am talking about testability and not actually testing is that I strongly believe the biggest benefit of writing tests is to write better and more maintainable code. As long as we strive for that, I personally don't really care too much about the test coverage. I mean, I still do, it's just not a big priority for me. In this case, if you look at `exists` method from another developer's perspective, if they don't actually look at the implementation of it, they won't even know that it's using the `FileManager` underneath and might even interpret this as a property that's set through the constructor or retrieved from network or something. (not necessarily for this feature of course, but in general)

Anyhow, that's a very long comment for something so small, but I wanted my first comment on this subject to explain my thought process. It's also not very well structured, but I hope it was easy enough to follow.

Let me know what you think!

Reply:
No objection here – that code is actually not called from anywhere at the moment anyway (it's for a future PR), so I'll refactor the rest of the code to remove it – it's easy enough to use the `LogFile`'s owner's `FileManager` to check if the file is around.

Reply:
This has been removed from this PR in fcb11cb509139e33e0a4fda70292f16c7925542b